kernelci.org: Add blog post about hackfests (#51)
kernelci/kernelci-project: kernelci.org/content/en/blog/posts/2021/hackfests.md

---
date: 2021-09-14
title: "KernelCI Hackfests"
linkTitle: "KernelCI Hackfests"
author: Guillaume Tucker
description: >
  Lessons learnt from the first two hackfests
---

KernelCI hackfests span over a few days during which a number of contributors

> get togher to focus on upstream Linux kernel testing.  So far, mainly kernel

Reviewer: ```suggestion
get together to focus on upstream Linux kernel testing.  So far, mainly kernel
```

participants.

## Timeline

Here's a summary of the first two hackfests:

### Hackfest #1 - 27th May to 4th June 2021

* Workboard: https://github.com/orgs/kernelci/projects/3
* Participans
  * Several kernel developers from [Google Chrome
  OS](https://www.google.com/intl/en_uk/chromebook/chrome-os/)
  * Several kernel developers from [Collabora](https://collabora.com)
  * A few members of the core KernelCI team
* Achievements
  * New KernelCI instance for Chrome OS on https://chromeos.kernelci.org
  * Added support for building Chrome OS configs on top of mainline
  * Enabled clang-13 builds for Chrome OS and main KernelCI builds
  * New test suite for libcamera enabled on https://linux.kernelci.org
  * Enabled LTP crypto tests with extra kernel config fragment
  * Several patches were also sent to extend kselftest and KUnit coverage in
      the kernel tree

### Hackfest #2 - 6th to 10th September 2021

* Workboard: https://github.com/orgs/kernelci/projects/5
* Participants
  * More members of the [TSC](/docs/org/tsc/) took part than first hackfest
  * New contributor: Denis Efremov (floppy disk kernel maintainer)

>   * Alice from [Cybertrust](https://www.cybertrust.co.jp/english/)

Reviewer 2: Cybertrust Japan

  * Chris from [Civil Infrastructure Platform](https://www.cip-project.org/)
    (CIP)

> * Achievements

Reviewer 2: Support for xilinx-zcu102 device

  OS](https://www.google.com/intl/en_uk/chromebook/chrome-os/)
  * Several kernel developers from [Collabora](https://collabora.com)
  * A few members of the core KernelCI team
* Achievements
  * New KernelCI instance for Chrome OS on https://chromeos.kernelci.org
  * Added support for building Chrome OS configs on top of mainline
  * Enabled clang-13 builds for Chrome OS and main KernelCI builds
  * New test suite for libcamera enabled on https://linux.kernelci.org
  * Enabled LTP crypto tests with extra kernel config fragment
  * Several patches were also sent to extend kselftest and KUnit coverage in
      the kernel tree

### Hackfest #2 - 6th to 10th September 2021

* Workboard: https://github.com/orgs/kernelci/projects/5
* Participants
  * More members of the [TSC](/docs/org/tsc/) took part than first hackfest
  * New contributor: Denis Efremov (floppy disk kernel maintainer)
  * Alice from [Cybertrust Japan](https://www.cybertrust.co.jp/english/)
  * Chris from [Civil Infrastructure Platform](https://www.cip-project.org/)
    (CIP)
  * Nikolai from [Red Hat's CKI](https://cki-project.org/)
  * Several KernelCI developers from [Collabora](https://collabora.com)

> * Achievements

Reviewer 2: CIP DNS request (cip.kernelci.org)

Author: OK, I'll phrase it as a redirect with the URLs

> * Achievements

Reviewer 2: Initial work for IEC cybersecurity tests

> * Achievements

Reviewer 2: Initial work for livepatch testing

Author: This is part of kselftest, right?

Reviewer 2: yes

---
date: 2021-09-20
title: "KernelCI Hackfests"
linkTitle: "KernelCI Hackfests"
author: Guillaume Tucker
description: >
  Lessons learned from the first two hackfests
---

KernelCI hackfests span over a few days during which a number of contributors
get together to focus on upstream Linux kernel testing.  So far, mainly kernel
and automated test system developers have been taking part in the hackfests but
anyone is welcome to join.  Topics mostly include extending test coverage in
various ways: enabling new test suites as well as adding test cases to
established frameworks such as kselftest and LTP, building additional kernel

> flavours, bringing up new particular types of hardware to be tested...

Reviewer 3: ```suggestion
flavours, bringing up new types of hardware to be tested...
```

> There have been two hackfests to this day.  The current plan is to hold one

Reviewer 3: ```suggestion
There have been two hackfests to date. The current plan is to hold one
```

Author: Ah yes, "to this day" is for something that continues in time.  The hackfests are over now!

## Connecting the dots

There is a large ecosystem around the Linux kernel which includes testing in
many shapes and forms: kernel developers, test developers, test system
developers, OEMs testing fully integrated products...  All these teams of
people don't necessarily interact with each other very much outside of their
organisations, and kernel developers aren't necessarily in the habit of writing
tests as part of their daily work.

Events such as the KernelCI hackfests give a chance for people from these
different areas to work together on solving common issues and keep the
ecosystem healthy.  It also helps with shifting the upstream Linux kernel
development culture towards a more test-driven workflow, to bring mainline
Linux closer to the real world where it is actually being used.

Let's consider a plausible hackfest story.  A first participant writes a new
test, say in kselftest.  A second participant enables the test to run in
KernelCI, which fails in some cases and a kernel bug is found.  A third
participant makes a fix for the bug, which can then be tested directly in
KernelCI to confirm it works as expected.  This may even happen on

> [staging.kernelci.org](https://staging.kernelci.org) before the patches for the

Reviewer 4: It might be obvious from the name, but I'd add a phrase explaining what staging is for us.

Author: Good point, I'll put a link to the documentation page about it.

participants.

## Timeline

Here's a summary of the first two hackfests:

### Hackfest #1 - 27th May to 4th June 2021

* Workboard: https://github.com/orgs/kernelci/projects/3
* Participans
  * Several kernel developers from [Google Chrome
  OS](https://www.google.com/intl/en_uk/chromebook/chrome-os/)
  * Several kernel developers from [Collabora](https://collabora.com)
  * A few members of the core KernelCI team
* Achievements
  * New KernelCI instance for Chrome OS on https://chromeos.kernelci.org
  * Added support for building Chrome OS configs on top of mainline
  * Enabled clang-13 builds for Chrome OS and main KernelCI builds
  * New test suite for libcamera enabled on https://linux.kernelci.org
  * Enabled LTP crypto tests with extra kernel config fragment
  * Several patches were also sent to extend kselftest and KUnit coverage in
      the kernel tree

### Hackfest #2 - 6th to 10th September 2021

* Workboard: https://github.com/orgs/kernelci/projects/5
* Participants
  * More members of the [TSC](/docs/org/tsc/) took part than first hackfest
  * New contributor: Denis Efremov (floppy disk kernel maintainer)

>   * Alice from [Cybertrust Japan](https://www.cybertrust.co.jp/english/)

Reviewer 3: And CIP?

Author: @aliceinwire Do you prefer to just have Cybertrust mentioned here, or CIP as well?

Reviewer 2: Both are ok

Reviewer 2: Sorry, please add both

>   * Chris from [Civil Infrastructure Platform](https://www.cip-project.org/)
>     (CIP)

Reviewer 3: Would it be rude to also add Renesas here?

```suggestion
  * Chris from [Civil Infrastructure Platform](https://www.cip-project.org/)
    (CIP) / [Renesas](https://www.renesas.com/eu/en)
```

  * Nikolai from [Red Hat's CKI](https://cki-project.org/)
  * Several KernelCI developers from [Collabora](https://collabora.com)
* Achievements
  * New Docker images and initial steps for running
      [`cvehound`](https://github.com/evdenis/cvehound)
  * On-going progress with Chrome OS tests using QEMU and hardware in LAVA
  * Support for CIP kernel configs
  * Initial work to enable CIP Core images (Debian based)
  * Set up redirect from [cip.kernelci.org](https://cip.kernelci.org) to the
    web dashboard page with [CIP jobs](https://linux.kernelci.org/job/cip/)
  * Support for `xilinx-zcu102` device
  * Initial work to enable IEC cybersecurity tests
  * Initial work to enable kselftest livepatch
  * Most of the content from https://foundation.kernelci.org replicated on the
    new [kernelci.org](https://kernelci.org) static site

>   * Started adding support for including kernel firmware files in test jobs

Reviewer 3: Ah I didn't realise this - fantastic.

There are some Renesas configs that need USB firmware to avoid a massive 60s timeout on boot...

Author: Yes, the plan is to have a list of firmware files in to include in the rootfs jobs initially in `rootfs-configs.yaml`.  So we'll pick a few to get the best size/usefulness ratio with the devices currently enabled in KernelCI.
